test/shared/RoutingTestHelpers.sol

```diff
@@ -107,4 +122,46 @@ contract RoutingTestHelpers is Test, Deployers {
         params.amountOut = uint128(amountOut);
         params.amountInMaximum = type(uint128).max;
     }
+
+    function _finalizeExecuteAndCheckSwap(
+        Currency inputCurrency,
+        Currency outputCurrency,
+        uint256 amountIn,
+        uint256 amountOut
+    ) internal {
+        _finalizeExecuteAndCheckSwap(inputCurrency, outputCurrency, amountIn, amountOut, false);
+    }
+
+    function _finalizeExecuteAndCheckSwap(
```

Comment:
this is a little hard to read/know whats being done and checked on each test. could you separate out the finalize/balance checkpoints, and the checkSwap part?

i dont think its really even needed to share the prevBalance checks/finalize stuff since its quite different (ie with eth stuff)

and then you can just have a function _checkSwap(curr0, curr1, prevBal0, prevBal1) and share that code?